framework/auth/campaigns.py

```diff
@@ -1,21 +1,27 @@
+from datetime import datetime, timedelta
 import furl
+import logging
 
 from modularodm import Q
 from modularodm.exceptions import NoResultsFound, QueryException, ImproperConfigurationError
 
 from website import mails
 from website.models import PreprintProvider
-from website.settings import DOMAIN
+from website.settings import DOMAIN, CAMPAIGN_REFRESH_THRESHOLD
 
 
 CAMPAIGNS = None
+CAMPAIGNS_LAST_REFRESHED = datetime.utcnow()
 
 
 def get_campaigns():
 
     global CAMPAIGNS
+    global CAMPAIGNS_LAST_REFRESHED
```

Comment:
I'm a liiiitle uncomfortable about the globals, but since this is a standalone function and not in an object, I guess it makes sense? Maybe make a todo and a ticket to look into refactoring this later.